scripts/validate-artifacts.ps1

```diff
@@ -34,142 +34,155 @@ Write-Output "Validating artifacts"
+        Write-Output "`u{2705} Checked if the Network Security Group resource exists - OK"
+    }  else { 
         Write-Output `u{1F914}
-        throw "Unable to verify that VM uses the public ssh key 'linuxboxsshkey'. Please make sure that in New-AzVm comandled, parameter '-SshKeyName' is set to the name of the public SSH key you created earlier, and that you are not setting the parameter '-GenerateSshKey'."
+        throw "More than one Network Security Group resource was found in the VM resource group. Please delete all un-used Network Security Group resources and try again."
     }
+} else {
+    Write-Output `u{1F914}
+    throw "Unable to find Network Security Group resouce. Please re-deploy the VM and try again."
+}
 
-    if ($virtualMachine.properties.storageProfile.imageReference.publisher -eq "canonical") { 
-        Write-Output "`u{2705} Checked Virtual Machine OS image publisher - OK" 
-    } else { 
-        Write-Output `u{1F914}
-        throw "Virtual Machine uses OS image from unknown published. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again."
-    }
-    if ($virtualMachine.properties.storageProfile.imageReference.offer.Contains('ubuntu-server') -and $virtualMachine.properties.storageProfile.imageReference.sku.Contains('22_04')) { 
-        Write-Output "`u{2705} Checked Virtual Machine OS image offer - OK"
-    } else { 
-        Write-Output `u{1F914}
-        throw "Virtual Machine uses wrong OS image. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again." 
-    }
+$sshNsgRule = ( $nsg.properties.securityRules | Where-Object { ($_.properties.destinationPortRange -eq '22') -and ($_.properties.access -eq 'Allow')} ) 
+if ($sshNsgRule)  {
+    Write-Output "`u{2705} Checked if NSG has SSH network security rule configured - OK"
+} else { 
+    Write-Output `u{1F914}
+    throw "Unable to fing network security group rule which allows SSH connection. Please check if you configured VM Network Security Group to allow connections on 22 TCP port and try again."
+}
 
-    if ($virtualMachine.properties.hardwareProfile.vmSize -eq "Standard_B1s") { 
-        Write-Output "`u{2705} Checked Virtual Machine size - OK"
-    } else { 
-        Write-Output `u{1F914}
-        throw "Virtual Machine size is not set to B1s. Please make sure that your script creates a VM with size B1s and try again."
-    }
+$httpNsgRule = ( $nsg.properties.securityRules | Where-Object { ($_.properties.destinationPortRange -eq '8080') -and ($_.properties.access -eq 'Allow')} ) 
+if ($httpNsgRule)  {
+    Write-Output "`u{2705} Checked if NSG has HTTP network security rule configured - OK"
+} else { 
+    Write-Output `u{1F914}
+    throw "Unable to fing network security group rule which allows HTTP connection. Please check if you configured VM Network Security Group to allow connections on 8080 TCP port and try again."
```

Comment:
There is a typo in the error message: 'fing' should be 'find'. This is a minor issue but could cause confusion.